ui/recieve_page.go

```diff
@@ -1 +1,317 @@
+											layout.Rigid(func() {
+												layout.Inset{Bottom: unit.Dp(5)}.Layout(win.gtx, func() {
+													win.outputs.selectedWalletNameLabel.Layout(win.gtx)
+												})
+											}),
+											layout.Rigid(func() {
+												layout.Inset{Left: unit.Dp(22)}.Layout(win.gtx, func() {
+													win.outputs.selectedWalletBalLabel.Layout(win.gtx)
+												})
+											}),
+										)
+									})
+								}),
+							)
+						})
+					}
+					decredmaterial.Card{}.Layout(win.gtx, selectedDetails)
+				}),
+			)
+		}),
+	)
+}
+
+func (win *Window) qrCodeAddressColumn() {
+	qrCode, err := qrcode.New(addrs, qrcode.Highest)
+	if err != nil {
+		win.outputs.err.Text = err.Error()
+		return
+	}
+
+	qrCode.DisableBorder = true
+	layout.Flex{Axis: layout.Vertical}.Layout(win.gtx,
+		layout.Rigid(func() {
+			layout.Inset{Top: unit.Dp(16)}.Layout(win.gtx, func() {
+				layout.Flex{}.Layout(win.gtx,
+					layout.Flexed(0.2, func() {
+					}),
+					layout.Flexed(1, func() {
+						img := win.theme.Image(paint.NewImageOp(qrCode.Image(520)))
+						img.Src.Rect.Max.X = 521
+						img.Src.Rect.Max.Y = 521
+						img.Scale = 0.5
+						img.Layout(win.gtx)
+					}),
+				)
+			})
+		}),
+		layout.Rigid(func() {
+			layout.Inset{Top: unit.Dp(16)}.Layout(win.gtx, func() {
+				layout.Flex{}.Layout(win.gtx,
+					layout.Flexed(0.1, func() {
+					}),
+					layout.Flexed(1, func() {
+						win.receiveAddressColumn()
+					}),
+				)
+			})
+		}),
+	)
+}
+
+func (win *Window) receiveAddressColumn() {
+	layout.Flex{}.Layout(win.gtx,
+		layout.Rigid(func() {
+			win.outputs.receiveAddressLabel.Text = addrs
+			win.outputs.receiveAddressLabel.Layout(win.gtx)
+		}),
+		layout.Rigid(func() {
+			layout.Inset{Left: unit.Dp(16)}.Layout(win.gtx, func() {
+				for win.inputs.copy.Clicked(win.gtx) {
+					clipboard.WriteAll(addrs)
+					win.addressCopiedLabel.Text = "Address Copied"
+					time.AfterFunc(time.Second*9, func() {
+						win.addressCopiedLabel.Text = ""
+					})
+				}
+				win.outputs.copy.Layout(win.gtx, &win.inputs.copy)
+			})
+		}),
+	)
+}
+
+func (win *Window) setDefaultPageValues() {
+	info := win.walletInfo.Wallets[win.selected]
+
+	for i := range info.Accounts {
+		win.setSelectedAccount(info, info.Accounts[i], false)
+		break
+	}
+}
+
+func (win *Window) setSelectedAccount(wallet wallet.InfoShort, account wallet.Account, generateNew bool) {
+	selectedWallet = &wallet
+	selectedAccount = &account
+
+	win.outputs.selectedAccountNameLabel.Text = account.Name
+	win.outputs.selectedWalletNameLabel.Text = wallet.Name
+	win.outputs.selectedWalletBalLabel.Text = dcrutil.Amount(account.SpendableBalance).String()
+	win.outputs.selectedAccountBalanceLabel.Text = wallet.Balance
+
+	var addr string
+	var err error
+
+	// create a new receive address everytime a new account is chosen
+	if generateNew {
+		addr, err = win.wallet.NextAddress(wallet.ID, account.Number)
+		if err != nil {
+			win.outputs.err.Text = err.Error()
+			return
```

Comment:
Why do you need to generate a new address?

Reply:
you may need to be more explicit please as ion follow.

Reply:
Why do you need to call `NextAddress` every time a new account is chosen? The current one would remain unused then.

Reply:
so what do you recommend?